### PR TITLE
refactor(server): configure cyclone server by configmap

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

configure cyclone server by configmap instead of flags

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -39,34 +40,28 @@ import (
 
 // APIServerOptions contains all options(config) for api server
 type APIServerOptions struct {
```
Comment (cd1989) [inline cmd/server/main.go]: I'd prefer name `Options` as it's in `server` package.

File: cmd/server/main.go [hunk 2]
```diff
@@ -39,34 +40,28 @@ import (
...
-	CycloneAddr string
-
-	Loglevel string
-
-	// StorageClass is used to create pvc for default tenant
-	StorageClass string
+	// ConfigMap that configures for cyclone server
+	ConfigMap string
+	// Namespace that cyclone server will run in
+	Namespace string
```
Comment (cd1989) [inline cmd/server/main.go]: I think these two items should have default values, please describe them here.

File: pkg/server/config/config.go [hunk 3]
```diff
@@ -1,130 +1,83 @@
...
+	if config.CycloneServerHost == "" {
+		log.Warningf("CycloneServerHost not configured, cyclone server can not listen API requests.")
 	}
 
-	intVal, err := strconv.ParseInt(val, 10, 32)
-	if err != nil {
-		log.Fatalf("The value '%s' of environment variable '%s' is not a valid integer: %v\n", val, name, err)
-		os.Exit(-1)
+	if config.CycloneServerPort == 0 {
+		log.Warningf("CycloneServerPort not configured, cyclone server can not listen API requests.")
```
Comment (cd1989) [inline pkg/server/config/config.go]: If `cyclone server can not listen API requests`, should return `false` in my understand. And I think here we should fallback to default value. The same to server host, these two config items should be optional for users.

Comment (zhujian7) [reply pkg/server/config/config.go]: OK

**Comment by zhujian7** [comment]
@cd1989  All comments have been fixed, PTAL

**Pull request merged**